### DISCUSSION TITLE
Automatically Lock Doors After Set Period/Location Change of Phone

### DISCUSSION THREAD

**Discussion by IIAIronWolf** [G-Two/homeassistant-subaru#34] [Ideas]
I'm wanting to write an automation with this integration that will allow me to automatically lock my doors (something Subaru should have added as a factory setting -____- ) after a set period of time with the ignition off, or if my phone location changes from my cars location. I'm guessing this will require templating, which I am not very familiar with. 

Has anyone done something like this or have any suggestions?

**Comment by stboch** [comment]
I have an automation that automatically locks my car at 10pm and 12am every night. 

I use scripts for everything so just create a script for the locks, then you can easily change all your automation by just editing one script.

It is really easy to add triggers and conditions so that when your car device_tracker updates use a timer or condition. I would also recommend using the Bluetooth connection data from the home-assistant app to identify when you leave your car.

**Reply by IIAIronWolf** [reply]
Would you mind sharing some examples? I haven't written a script in so long and I'm not even sure where to start. Haha

**Reply by IIAIronWolf** [reply]
Also, unless I'm mistaken, there is no bluetooth sensor for iOS devices, which is what I use. :/

**Reply by stboch** [reply]
Scripts like this are really simple can be done in the UI. here is my lock script, basically just go to automations/scripts/scenes and add new script on the script tab. then at the bottom for actions choose the device that is your car and the lock that goes with it. Then you can use it in automations by using the call service and then choose the script will be called script.(whatever_you_named_it)

Example Screenshot:
![Screenshot_20220103-043721_Home Assistant.jpg](https://user-images.githubusercontent.com/7524887/147916509-8845585f-25eb-44d1-83cd-f4cf8c7bf0a7.jpg)

**Reply by stboch** [reply]
> Also, unless I'm mistaken, there is no bluetooth sensor for iOS devices, which is what I use. :/

since I only use Android I didn't realize that this is isn't a supported feature on iOS shame really it's super easy to use.
